src/hooks/useSelectionLogic.ts

```diff
@@ -142,23 +139,35 @@ export function useSelectionLogic<T extends HTMLElement>({
   );
 
   const onMouseMove = useCallback(
-    (e: MouseEvent) => {
+    (event: MouseEvent) => {
       if (!startPoint.current) {
         return;
       }
 
       const rect = containerRef.current?.getParentBoundingClientRect();
-      endPoint.current = getPointFromEvent(e, rect);
-      handleMouseMove(rect);
+      endPoint.current = getPointFromEvent(event, rect);
+      handleMouseMove(event, rect);
     },
     [handleMouseMove, getPointFromEvent, containerRef],
   );
 
   const onMouseUp = useCallback(
-    (e: MouseEvent) => {
-      // handle only left button click
-      if (e.button === 0) {
+    (event: MouseEvent) => {
+      /**
+       * handle only left button up event
+       */
+      if (event.button === 0) {
+        /**
+         * If the user just clicked down and up in the same place without dragging,
+         * we don't want to fire the onSelectionEnd event. We can do this
+         * by checking if endPoint.current exists.
+         */
+        if (endPoint.current) {
+          currentSelectionEnd.current?.(event);
+        }
```

Comment:
the main fix here was pulling `currentSelectionEnd` out of `cancelCurrentSelection` and conditionally calling it